pkg/network/ovn_kubernetes.go

```diff
@@ -212,6 +215,26 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
 	return objs, nil
 }
 
+// returns the value of mode found in the openshift-ovn-kubernetes/gateway-mode-config configMap
+// if it exists, otherwise returns whatever the global OVN_GATEWAY_MODE is set to (shared)
+func GetGatewayMode(kubeClient client.Client) string {
+	cm := &corev1.ConfigMap{
+		TypeMeta:   metav1.TypeMeta{APIVersion: "v1", Kind: "ConfigMap"},
```

Comment:
FYI: you don't need to set the TypeMeta (though it doesn't hurt anything)

Reply:
In fact, this can be a zero-valued object (there were a few exceptions due to bugs with CRD and type registration, but they're all gone now).